server/auth_token_test.go

```diff
@@ -0,0 +1,160 @@
+package main
+
+import (
+	"crypto/aes"
+	"crypto/cipher"
+	"encoding/json"
+	"errors"
+	"io"
+	"testing"
+
+	"bou.ke/monkey"
+	"github.com/stretchr/testify/assert"
+	"golang.org/x/oauth2"
+)
+
+type mockBLock struct{}
+
+func (b *mockBLock) BlockSize() int { return 0 }
+
+func (b *mockBLock) Encrypt(_, _ []byte) {}
+
+func (b *mockBLock) Decrypt(_, _ []byte) {}
+
+type mockAesgcm struct{}
+
+func (a *mockAesgcm) NonceSize() int { return 1 }
+
+func (a *mockAesgcm) Overhead() int { return 1 }
+
+func (a *mockAesgcm) Seal(dst, nonce, plaintext, additionalData []byte) []byte { return []byte("mock") }
+
+func (a *mockAesgcm) Open(dst, nonce, ciphertext, additionalData []byte) ([]byte, error) {
+	return []byte("mock"), nil
+}
+
+func Test_NewEncodedAuthToken(t *testing.T) {
+	defer monkey.UnpatchAll()
+	for _, testCase := range []struct {
+		description    string
+		expectedError  string
+		newCipherError error
+		newGCMError    error
+		readFullError  error
+	}{
+		{
+			description: "NewEncodedAuthToken: oAuth token is encoded successfully",
+		},
+		{
+			description:    "NewEncodedAuthToken: failed to create oAuth token because aes.NewCipher gives error",
```

Comment:
Check this everywhere
```suggestion
			description:    "NewEncodedAuthToken: failed to create the oAuth token because aes.NewCipher gives error",
```